contracts/RepaymentController.sol

```diff
@@ -152,30 +192,62 @@ contract RepaymentController is IRepaymentController, InterestCalculator, FeeLoo
      * @dev Shared logic to perform validation and calculations for repay and forceRepay.
      *
      * @param loanId               The ID of the loan.
+     * @param amount               The amount to repay.
      *
-     * @return amountFromBorrower   The amount to collect from the borrower.
      * @return amountToLender       The amount owed to the lender.
+     * @return interestAmount       The amount of interest due.
+     * @return paymentToPrincipal   The portion of the repayment amount that goes to principal.
      */
-    function _prepareRepay(uint256 loanId) internal view returns (uint256 amountFromBorrower, uint256 amountToLender) {
+    function _prepareRepay(uint256 loanId, uint256 amount)
+        internal
+        view
+        returns (
+            uint256 amountToLender,
+            uint256 interestAmount,
+            uint256 paymentToPrincipal
+        )
+    {
+        // get loan data
         LoanLibrary.LoanData memory data = loanCore.getLoan(loanId);
+
+        // loan state checks
         if (data.state == LoanLibrary.LoanState.DUMMY_DO_NOT_USE) revert RC_CannotDereference(loanId);
         if (data.state != LoanLibrary.LoanState.Active) revert RC_InvalidState(data.state);
 
-        LoanLibrary.LoanTerms memory terms = data.terms;
-
-        uint256 interest = getProratedInterestAmount(
+        // get interest amount due
+        interestAmount = getProratedInterestAmount(
             data.balance,
-            terms.interestRate,
-            terms.durationSecs,
+            data.terms.interestRate,
+            data.terms.durationSecs,
             uint64(data.startDate),
             uint64(data.lastAccrualTimestamp),
             block.timestamp
         );
 
-        uint256 interestFee = (interest * data.feeSnapshot.lenderInterestFee) / BASIS_POINTS_DENOMINATOR;
-        uint256 principalFee = (terms.principal * data.feeSnapshot.lenderPrincipalFee) / BASIS_POINTS_DENOMINATOR;
-
-        amountFromBorrower = terms.principal + interest;
+        if(amount == MAX_INT) {
```

Comment:
Can remove this (along with the defined constant). Won't lines 238-240 already handle this cleanly?